docs/src/architecture/state.md

```diff
@@ -1,74 +1,86 @@
+| **Zero-knowledge helps with** | Transactions executed only once | No need to know full state to validate blocks |
+| **Concurrent state model helps with** | Transactions executed concurrently by distinct actors | No need to kow full state to produce blocks |
+
+
+## State components
 The Miden Node(s) maintain three databases to describe the state:
 1. A database of accounts.
 2. A database of notes.
 3. A database of nullifiers for already consumed notes.
 
 <p align="center">
-  <img src="../diagrams/architecture/state/State.png">
+  <img src="../diagrams/architecture/state/State.png" width="80%">
 </p>
 
-## State components
-
 These databases are represented by authenticated data structures, such that we can easily prove that items were added to or removed from a database, and a commitment to the database would be very small.
 
-### Account database
-Current account states are recorded in a Sparse Merkle tree which maps account IDs to account hashes, where the account hash is computed as
+Polygon Miden has two databases to capture the note states. The note database is append-only and permanently stores all notes, while the nullifier database stores which notes cannot be consumed anymore. Separating note storage into these two databases gives Polygon Miden client-side proving and advanced privacy.
 
-`hash([account ID, 0, 0, nonce], [vault root], [storage root], [code root])`.
+### Account database
+The latest account states are recorded in a [tiered sparse Merkle tree](https://0xpolygonmiden.github.io/miden-base/crypto-primitives/tsmt.html) which maps account IDs to account hashes and account data if needed. 
 
 <p align="center">
-  <img src="../diagrams/architecture/state/Account_DB.png">
+  <img src="../diagrams/architecture/state/Account_DB.png" width="80%">
 </p>
 
-As described in [accounts](https://0xpolygonmiden.github.io/miden-base/architecture/accounts.html), there are two types of accounts:
+As described in [Accounts](http://localhost:3000/architecture/accounts.html#account-storage-modes), there are three types of accounts:
 * **Public accounts** where all account data is stored on-chain.
+* **Encrypted accounts** where where the account data is stored onchain but in encrypted text.
 * **Private accounts** where only the hashes of accounts are stored on-chain.
 
-> Losing a state of a private account would mean loss of funds (as the user won't be able to execute transactions) in a similar manner as a loss of a private key would. This problem can be easily mitigated by storing encrypted account state in a cloud or backing it up somewhere else. Unlike storing private keys in the cloud, this does not compromise privacy or security of an account.
+Having many (or even most) of the accounts be private is very beneficial for the network as a private account contributes only $40$ bytes to the global state ($8$ bytes account ID + $32$ bytes account hash). Or, said another way, 1 billion private accounts takes up only $40$ GB of state.
```

Comment:
Private accounts significantly reduce the storage overhead for nodes. A private account contributes only $40$ bytes to the global state ($8$ bytes account ID + $32$ bytes account hash). Or, said another way, 1 billion private accounts takes up only $40$ GB of state.

Can we make a comparison with the size of on-chain accounts?

Reply:
done. Do you have the number for onchain accounts?

Reply:
Ah that depends on various things, like the code or the storage of the account.